app/pages/OrganizationEditPage.jsx

```diff
@@ -620,28 +620,38 @@ class OrganizationEditPage extends React.Component {
 
   handleDeactivation(type, id) {
     const { router } = this.props;
+    let confirmMessage = null;
+    let path = null;
+    if (type === 'resource') {
+      confirmMessage = 'Are you sure you want to deactive this resource?';
+      path = `/api/resources/${id}`;
+    } else if (type === 'service') {
+      confirmMessage = 'Are you sure you want to remove this service?';
+      path = `/api/services/${id}`;
+    }
     // eslint-disable-next-line no-restricted-globals
-    if (confirm('Are you sure you want to deactive this resource?') === true) {
-      let path = null;
-      if (type === 'resource') {
-        path = `/api/resources/${id}`;
-      } else if (type === 'service') {
-        path = `/api/services/${id}`;
-      }
-      dataService.APIDelete(path, { change_request: { status: '2' } })
-        .then(() => {
-          alert('Successfully deactivated! \n \nIf this was a mistake, please let someone from the ShelterTech team know.');
-          if (type === 'resource') {
-            // Resource successfully deactivated. Redirect to home.
-            router.push({ pathname: '/' });
-          } else {
-            // Service successfully deactivated. Mark deactivated in local state.
-            this.setState(state => {
-              state.deactivatedServiceIds.add(id);
-              return state;
-            });
-          }
+    if (confirm(confirmMessage) === true) {
+      if (id < 0) {
+        this.setState(state => {
+          state.deactivatedServiceIds.add(id);
```

Comment:
You're not supposed to directly mutate the values in a React `state` object, since React is unable to detect when those changes happen. Instead, React only wants you to ever modify state by calling the `setState()` method. See https://reactjs.org/docs/state-and-lifecycle.html#do-not-modify-state-directly

The subtle distinction here is that although you are calling `setState()` and returning the updated state, the way you are doing it involves directly changing the previous object.

The correct way of doing what you're trying to do here is to create a new array that contains the old `state.deactivatedServiceIds` array plus a new element at the end. Let me modify your code here to show what it should look like.

Reply:
Thanks @richardxia !